spec/controllers/archive_faqs_controller_spec.rb

```diff
@@ -6,38 +6,43 @@
   include LoginMacros
   include RedirectExpectationHelper
 
-  describe "GET #index" do
-    it "renders the index page anyway when the locale param is invalid" do
-      expect(I18n).to receive(:with_locale).with("eldritch").and_call_original
-      get :index, params: { language_id: "eldritch" }
-      expect(response).to render_template(:index)
-    end
+  let(:non_standard_locale) { create(:locale) }
+  let(:user_locale) { create(:locale) }
+  let(:user) do
+    user = create(:user)
+    user.preference.update!(preferred_locale: user_locale.id)
+    $rollout.activate_user(:set_locale_preference, user)
+    user
+  end
 
-    it "redirects to the default locale when the locale param is empty" do
-      expect(I18n).not_to receive(:with_locale)
-      get :index, params: { language_id: "" }
-      it_redirects_to(archive_faqs_path(language_id: I18n.default_locale))
-    end
+  describe "GET #index" do
+    context "when there's no locale in session" do
+      it "redirects to the default locale when the locale param is invalid" do
+        expect(I18n).not_to receive(:with_locale)
+        get :index, params: { language_id: "eldritch" }
+        it_redirects_to(archive_faqs_path(language_id: I18n.default_locale))
+      end
 
-    it "redirects to the default locale when the locale param is empty and the session locale is not" do
-      expect(I18n).not_to receive(:with_locale)
-      get :index, params: { language_id: "" }, session: { language_id: "eldritch" }
-      it_redirects_to(archive_faqs_path(language_id: "en"))
-    end
+      it "redirects to the default locale when the locale param is empty" do
+        expect(I18n).not_to receive(:with_locale)
+        get :index, params: { language_id: "" }
+        it_redirects_to(archive_faqs_path(language_id: I18n.default_locale))
+      end
 
-    it "redirects to the default locale when the locale param and the session locale are empty" do
-      expect(I18n).not_to receive(:with_locale)
-      get :index, params: { language_id: "" }, session: { language_id: "" }
-      it_redirects_to(archive_faqs_path(language_id: "en"))
+      it "redirects to the default locale when the locale param and the session locale are empty" do
```

Comment:
Isn't this redundant with `redirects to the default locale when the locale param is empty`, since the context is that `session[:locale]` is unset?

Reply:
This test, as is, is useful to catch an edge case forcing to add a couple of `.presence`, the one where `session[:language_id]` is blank (nil actually), which isn't something possible with the new code, but used to be with the old one. So it's kinda a legacy failsafe.

Name should definitely be updated to reflect that though.

Reply:
Ah, gotcha. Feel free to mark this thread as resolved too